service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerTest.java

```diff
@@ -105,30 +105,60 @@ void aboutToStartShouldReturnErrorsWhenFamilymanNumberIsNotProvided() throws Exc
     void midEventShouldGenerateOrderDocument() throws Exception {
         byte[] pdf = {1, 2, 3, 4, 5};
         Document document = document();
-        DocmosisDocument docmosisDocument = new DocmosisDocument(C21.getDocumentTitle(), pdf);
+        DocmosisDocument docmosisDocument = new DocmosisDocument("order.pdf", pdf);
 
         given(docmosisDocumentGeneratorService.generateDocmosisDocument(any(), any())).willReturn(docmosisDocument);
-        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, C21.getDocumentTitle()))
+        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, "blank_order_c21.pdf"))
             .willReturn(document);
 
         AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(callbackRequest(), "mid-event");
 
         CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
 
-        assertThat(caseData.getOrder().getDocument()).isEqualTo(DocumentReference.builder()
+        assertThat(caseData.getOrderTypeAndDocument().getDocument()).isEqualTo(DocumentReference.builder()
             .binaryUrl(document.links.binary.href)
             .filename(document.originalDocumentName)
             .url(document.links.self.href)
             .build());
     }
 
     @Test
-    void aboutToSubmitShouldUpdateCaseData() throws Exception {
+    void aboutToSubmitShouldUpdateCaseDataAccordinglyWhenC21IsSelected() throws Exception {
```

Comment:
Don't like these shouldUpdateCaseData, surely we could be more specific as to what these tests are doing?

Reply:
This is a test that asserts that an order of type C21 got added to case data, the OrderTypeAndDocument object got removed from case data, the Order object got removed from case data and the JudgeAndLegalAdvisor object got removed from case data. Basically asserts that CaseData is in the state we expect it to be in at the end of the test. If you have a better name than update then let me know 👍

Reply:
change name for you